src/cascadia/TerminalSettingsEditor/Actions.h

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "Actions.g.h"
+#include "ActionsPageNavigationState.g.h"
+#include "Utils.h"
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    struct ActionsPageNavigationState : ActionsPageNavigationStateT<ActionsPageNavigationState>
+    {
+    public:
+        ActionsPageNavigationState(const Model::CascadiaSettings& settings) :
+            _Settings{ settings } {}
+
+        void RequestOpenJson()
+        {
+            _OpenJsonHandlers(nullptr, winrt::Microsoft::Terminal::Settings::Model::SettingsTarget::SettingsFile);
+        }
+
+        GETSET_PROPERTY(Model::CascadiaSettings, Settings, nullptr)
+        TYPED_EVENT(OpenJson, Windows::Foundation::IInspectable, Model::SettingsTarget);
+    };
+
+    struct Actions : ActionsT<Actions>
+    {
+    public:
+        Actions();
+
+        void OnNavigatedTo(const winrt::Windows::UI::Xaml::Navigation::NavigationEventArgs& e);
+
+        Windows::Foundation::Collections::IObservableVector<winrt::Microsoft::Terminal::Settings::Model::Command> FilteredActions();
+
+        GETSET_PROPERTY(Editor::ActionsPageNavigationState, State, nullptr);
+
+    private:
+        friend struct ActionsT<Actions>; // for Xaml to bind events

```

Comment:
_Weird!_ Is this a trick from the CommandPalette? Why didn't I have to do any of this before?

Reply:
We have to do this for most of the XAML-like things - any time we bind to a `private` method.